libs/providers/vmware.py

```
@@ -1268,6 +1268,26 @@ def clone_vm(
             else:
                 LOGGER.warning("Disk type '%s' not recognized. Using vSphere default.", disk_type)
 
+            if disk_type.lower() == "thick-eager":
+                disk_locators: list[vim.vm.RelocateSpec.DiskLocator] = []
+                for device in source_vm.config.hardware.device:
+                    if not isinstance(device, vim.vm.device.VirtualDisk):
+                        continue
+                    if not isinstance(device.backing, vim.vm.device.VirtualDisk.FlatVer2BackingInfo):
+                        LOGGER.debug(
+                            f"Skipping non-flat disk '{device.deviceInfo.label}' for eager scrub configuration."
```

Ruff (0.15.11)

[warning] Logging statement uses f-string

(G004)